telegram/_payment/stars/transactionpartner.py

```diff
@@ -281,55 +281,107 @@ class TransactionPartnerUser(TransactionPartner):
     """Describes a transaction with a user.
 
     Objects of this class are comparable in terms of equality. Two objects of this class are
-    considered equal, if their :attr:`user` are equal.
+    considered equal, if their :attr:`user` and :attr:`transaction_type` are equal.
 
     .. versionadded:: 21.4
 
+    .. versionchanged:: NEXT.VERSION
+       Equality comparison now includes the new required argument :paramref:`transaction_type`,
+       introduced in Bot API 9.0.
+
     Args:
+        transaction_type (:obj:`str`): Type of the transaction, currently one of
+            :tg-const:`telegram.constants.TransactionPartnerUser.INVOICE_PAYMENT` for payments via
+            invoices, :tg-const:`telegram.constants.TransactionPartnerUser.PAID_MEDIA_PAYMENT`
+            for payments for paid media,
+            :tg-const:`telegram.constants.TransactionPartnerUser.GIFT_PURCHASE` for gifts sent by
+            the bot, :tg-const:`telegram.constants.TransactionPartnerUser.PREMIUM_PURCHASE`
+            for Telegram Premium subscriptions gifted by the bot,
+            :tg-const:`telegram.constants.TransactionPartnerUser.BUSINESS_ACCOUNT_TRANSFER` for
+            direct transfers from managed business accounts.
```

Comment:
```suggestion
            direct transfers from managed business accounts.
            
            .. versionadded:: NEXT.VERSION
```

same in the attributes list :)